go/pipeline/base_batch.go

```diff
@@ -0,0 +1,6454 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package pipeline
+
+// #include "../lib.h"
+import "C"
+
+import (
+	"fmt"
+	"math"
+	"reflect"
+	"strconv"
+
+	"github.com/valkey-io/valkey-glide/go/v2/constants"
+	"github.com/valkey-io/valkey-glide/go/v2/models"
+	"github.com/valkey-io/valkey-glide/go/v2/options"
+
+	"github.com/valkey-io/valkey-glide/go/v2/internal"
+	"github.com/valkey-io/valkey-glide/go/v2/internal/errors"
+	"github.com/valkey-io/valkey-glide/go/v2/internal/utils"
+)
+
+// Executes a single command, specified by args, without checking inputs. Every part of the command,
+// including the command name and subcommands, should be added as a separate value in args. The returning value depends on
+// the executed command.
+//
+// See [Valkey GLIDE Wiki] for details on the restrictions and limitations of the custom command API.
+//
+// This function should only be used for single-response commands. Commands that don't return complete response and awaits
+// (such as SUBSCRIBE), or that return potentially more than a single response (such as XREAD), or that change the client's
+// behavior (such as entering pub/sub mode on RESP2 connections) shouldn't be called using this function.
+//
+// Parameters:
+//
+//	args - Arguments for the custom command.
+//
+// Command Response:
+//
+//	The returned value for the custom command.
```

Comment:
The `Command Response` docs seem a little misleading. Each command returns `*T`. It is the `executeBatch` that returns values. Maybe there should be a preamble like "Command Responses are returned when the Batch is executed."

Reply:
We should apply this change to all clients - we can do this in another PR.